### PR TITLE
ITOps consolidation

### LINKED ISSUE #80: Consolidate infrastructure/itOps projects

### PR DESCRIPTION
**Pull request by tommasobertoni**
- introduces BlackClientFactory
- provides already-initialized BlackClient
- use of shared itops project: artiso.AdsdHotel.ITOps
- use of IMessageConvention

Next steps:
- remove `artiso.AdsdHotel.Infrastructure.NServiceBus` project (unused)
- move `artiso.AdsdHotel.ITOps` project in `services/infrastructure/src`.
- consolidate "data storage" projects (move everything in `artiso.AdsdHotel.ITOps` or provide different projects?)

**Note:** I am not providing different callback configurations for the client and the api, in order to make the methods simpler.
If there are important implications with this, we may add a flag "make calls" in the `EndpointConfiguration Create(..` method.

Resolves #80.

### CODE REVIEW

File: services/black/src/ambassador/BlackClientFactory.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public BlackClientFactory(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public async Task<BlackClient> CreateAsync()
+        {
+            var endpointConfiguration = NServiceBusEndpointConfigurationFactory.Create(
+                endpointName: $"Black.Client.{Guid.NewGuid()}",
+                rabbitMqConnectionString: "host=localhost");
```
Comment (tziemek) [inline services/black/src/ambassador/BlackClientFactory.cs:22]: We can use the factory, but we should change the factory implementation. Right know it's always with callbacks etc. In the infrastructure project a split these, so we are more flexible.

File: services/blue/src/itops/Abstraction/NServiceBus/AdsdHotelMessageConventions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using artiso.AdsdHotel.ITOps.Communication;
+using NServiceBus;
+
+namespace artiso.AdsdHotel.ITOps.Abstraction.NServiceBus
+{
+    internal class AdsdHotelMessageConventions : IMessageConvention
```
Comment (tziemek) [inline services/blue/src/itops/Abstraction/NServiceBus/AdsdHotelMessageConventions.cs:7]: Reuse the class in the infrastructure project.

**Review by tziemek — COMMENTED** [review]
We should merge ITOPS with infrastructure. I don't like a single project for ITOPS because I don't want to reference some SQL stuff only because I want to use the channel abstraction.

File: services/blue/src/itops/Abstraction/NServiceBus/AdsdHotelMessageConventions.cs (hunk 2 above)
Comment (tommasobertoni) [reply services/blue/src/itops/Abstraction/NServiceBus/AdsdHotelMessageConventions.cs:7]: With the consolidation of the projects, only one `AdsdHotelMessageConventions` will remain.

File: services/black/src/ambassador/BlackClientFactory.cs (hunk 1 above)
Comment (tommasobertoni) [reply services/black/src/ambassador/BlackClientFactory.cs:22]: Feel free to implement the flag in the newly-consolidated ITOps project.

**Review by tziemek — APPROVED** [review]

**Review by antonio8101 — APPROVED** [review]

**Pull request merged**